pkg/controller/master/image/vm_image_controller.go

```diff
@@ -397,3 +399,45 @@ func handleFail(image *harvesterv1.VirtualMachineImage, cond condition.Cond, err
 	}
 	return image
 }
+
+func (h *vmImageHandler) deleteVMImageMetadata(image *harvesterv1.VirtualMachineImage) error {
+	target, err := settings.DecodeBackupTarget(settings.BackupTargetSet.Get())
+	if err != nil {
+		return err
+	}
+
+	// when backup target has been reset to default, skip following
+	if target.IsDefaultBackupTarget() {
+		logrus.WithFields(logrus.Fields{
+			"namespace": image.Namespace,
+			"name":      image.Name,
+		}).Debugf("skip deleting vm image metadata, because backup target setting is default")
+		return nil
+	}
+
+	if !util.IsBackupTargetSame(image.Status.BackupTarget, target) {
+		logrus.WithFields(logrus.Fields{
+			"namespace":            image.Namespace,
+			"name":                 image.Name,
+			"status.backupTarget":  image.Status.BackupTarget,
+			"setting.backupTarget": target,
+		}).Debugf("skip deleting vm image metadata, because status backup target is different from backup target setting")
+		return nil
+	}
+
+	bsDriver, err := util.GetBackupStoreDriver(h.secretCache, target)
+	if err != nil {
+		return err
+	}
+
+	destURL := util.GetVMImageMetadataFilePath(image.Namespace, image.Name)
+	if exist := bsDriver.FileExists(destURL); exist {
+		logrus.WithFields(logrus.Fields{
+			"namespace":    image.Namespace,
+			"name":         image.Name,
+			"backupTarget": image.Status.BackupTarget,
+		}).Debugf("delete vm image metadata in backup target")
+		return bsDriver.Remove(destURL)
+	}
+	return nil
+}
```

Comment:
I'm not sure about this: if we delete the VMImage metadata on the remote target, will this behavior contradict the idea of backing up an image?

Reply:
Hi @WebberHuang1118, thanks for the review. I think removing VMImage metadata is a safe operation with PR https://github.com/harvester/harvester/pull/6786, because VMImage can't be removed if there is any VMBackup using it. That means if VMImage can be removed, there is no related VMBackup in the backup target.